addons/tacticalladder/functions/fnc_cancelTLdeploy.sqf

```diff
@@ -32,7 +32,6 @@ GVAR(ladder) animate ["rotate", 0];
 
 {
     GVAR(ladder) animate [_x, 0];
-    true
```

Comment:
https://github.com/acemod/ACE3/pull/4815
This was added by you to fix some mcc thing, is this no longer needed?

Reply:
Actually looking back at it.. I don't know how that's supposed to have fixed that bug.
See https://github.com/acemod/ACE3/issues/4814#issue-200180402 
>Second animate should always return Nothing according to BIKI so the error should always be triggered. I don't know why it sometimes apparently returns something that count accepts as a bool.

The `true` was added to make the `count` not error when anything non-bool got to it. Replacing the count by a forEach made it faster (because we don't need the call to `true`) and well.. removed the need for the `true`.

And for whatever reason `animate` would return crap when fed with a nullObj 
https://github.com/acemod/ACE3/issues/4814#issuecomment-272030822
This just removes all constraints on the return values.

Reply:
Only problem now is.. That I didn't actually swap out the `count` for a `forEach` 🤔

Reply:
We probably should fix the core issue instead. Right-clicking while not currently deploying a ladder, shouldn't execute this function. Maybe that was fixed already?
If we fix the underlying issue, `ladder` will not be nil/null here and we won't have this problem.

Reply:
https://github.com/acemod/ACE3/search?q=cancelTLdeploy&unscoped_q=cancelTLdeploy

https://github.com/acemod/ACE3/blob/e5a15d200f44df5fccc0bc5575d18d80b35538dd/addons/tacticalladder/XEH_missionDisplayLoad.sqf#L6
This seems to be the problem. The rightclick and scrollwheel handlers are always registered. Instead of only when you're trying to place a ladder.

Executed from https://github.com/acemod/ACE3/blob/e5a15d200f44df5fccc0bc5575d18d80b35538dd/addons/tacticalladder/CfgEventHandlers.hpp#L30 at mission start.

Reply:
@jonpas added that in https://github.com/acemod/ACE3/pull/3524 

@commy2 already found the problem back then https://github.com/acemod/ACE3/pull/3524#issuecomment-192816064
>previously these actions would only be called when the current action was active (e.g. carrying a ladder), but now all of them are executed when you RMB every time.


Seems like `explosives` and `sandbag` have the same underlying problem, but they are not affected because they exit if the placement PFH is not running.


https://github.com/acemod/ACE3/pull/3524/files#diff-963d8341c69f48d73203fbd72841ef32R23 This check fails. Because the ladder is nil, causing the isNull check to return nil. Causing a `if (nil) exitWith {}` causing a `nil exitWith {}` and the exitWith also being ignored because of the nil parameter.
So either add a isNil check there, or stop always adding the EH and only add it when it's needed.
Should be fairly simple as sandbag/explosives/trenches all check for their placement PFH running, so we can just add the EH when the placement PFH starts, and remove it when it ends.

Reply:
I don't see anyway for `GVAR(ladder)` to become nil. 
Was it mcc messing with things?
I'd say just swap back to `forEach`, performance of this code does not matter

Reply:
`performance of this code does not matter` it's executed every time the player makes a right-click. :U 
But if postInit runs properly. It will exit at the isNull check at the start of the file.. So maybe not thaat bad. But the problem of the unused eventhandlers always being added still stands.